system/mp/mpcomm.cpp

```diff
@@ -831,13 +832,23 @@ protected: friend class CMPPacketReader;
 #ifdef _TRACE
                                 EXCLOG(e, "MP: Failed to connect");
 #endif
+                                // if other side closes because of stale connection on same host:port, connect again
+                                if (e->errorCode() == JSOCKERR_graceful_close)
```

Comment:
This change makes sense afaics, but don't see how related to epoll, so should be separate commit I think.